[CI] LookupJoinTypesIT testLookupJoinOthers failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #10249 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/cd6yj66okzxru)
- [elasticsearch-pull-request #87730 / part-3](https://gradle-enterprise.elastic.co/s/bwhon3ygep554)
- [elasticsearch-pull-request #87581 / part-3](https://gradle-enterprise.elastic.co/s/s34geysc4m6gk)
- [elasticsearch-pull-request #87478 / part-3](https://gradle-enterprise.elastic.co/s/wuizi6p74buwm)
- [elasticsearch-pull-request #86924 / part-3](https://gradle-enterprise.elastic.co/s/yvt3wcu7ukdwi)
- [elasticsearch-pull-request #86801 / part-3](https://gradle-enterprise.elastic.co/s/knaznfts3yjyq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.LookupJoinTypesIT.testLookupJoinOthers" -Dtests.seed=17F68441DB0C201F -Dtests.locale=rwk -Dtests.timezone=America/Guadeloupe -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.LookupJoinTypesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testLookupJoinOthers'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [elasticsearch[main-cluster-1][search][T#2]][testStopQueryLocal]
	org.elasticsearch.xpack.esql.plugin.ComputeListener.acquireAvoid(ComputeListener.java:45)
	org.elasticsearch.xpack.esql.plugin.ComputeListener.acquireCompute(ComputeListener.java:59)
	org.elasticsearch.xpack.esql.plugin.DataNodeComputeHandler$DataNodeRequestExecutor$1.<init>(DataNodeComputeHandler.java:266)
	org.elasticsearch.xpack.esql.plugin.DataNodeComputeHandler$DataNodeRequestExecutor.runBatch(DataNodeComputeHandler.java:265)
	org.elasticsearch.xpack.esql.plugin.DataNodeComputeHandler$DataNodeRequestExecutor.start(DataNodeComputeHandler.java:255)
	org.elasticsearch.xpack.esql.plugin.DataNodeComputeHandler.runComputeOnDataNode(DataNodeComputeHandler.java:441)
	org.elasticsearch.xpack.esql.plugin.DataNodeComputeHandler.messageReceived(DataNodeCo
[truncated]
```

**Issue Reasons:**
- [main] 6 failures in test testLookupJoinOthers (0.6% fail rate in 982 executions)
- [main] 5 failures in step part-3 (1.2% fail rate in 402 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (1.3% fail rate in 399 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -474,9 +474,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlClientYamlIT
   method: test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version}
   issue: https://github.com/elastic/elasticsearch/issues/133461
-- class: org.elasticsearch.xpack.esql.action.LookupJoinTypesIT
-  method: testLookupJoinOthers
-  issue: https://github.com/elastic/elasticsearch/issues/133480
 - class: org.elasticsearch.xpack.esql.action.CrossClusterAsyncQueryStopIT
   method: testStopQueryLocal
   issue: https://github.com/elastic/elasticsearch/issues/133481

